src/module.c

```diff
@@ -60,6 +71,54 @@ int _RegisterDataTypes(RedisModuleCtx *ctx) {
 	return REDISMODULE_OK;
 }
 
+static void _PrepareModuleGlobals() {
+	assert(pthread_mutex_init(&_module_mutex, NULL) == 0);
+	graphs_in_keyspace = array_new(GraphContext *, 1);
+	process_is_child = false;
+}
+
+static void RG_ForkPrepare() {
+	/* At this point, a fork call has been issued. (We assume that this is because BGSave was called.)
+	 * Acquire the writer mutex of each graph to ensure that no locks are held, or else
+	 * the child process will deadlock when attempting to acquire that lock.
```

Comment:
is this still true? we're acquiring the write locks to make sure we're in a consistent state.

Reply:
Improving the comment, but it's actually better now, since the writers' mutex causes far longer waits than acquiring the read-write lock as a reader (which is all we need).

If a bgsave is invoked while a query like `MATCH...CREATE` is in progress, the snapshot may be taken before or after that query is committed, but the data integrity of the snapshot is guaranteed. I think this is valid behavior for a bgsave!